### PR TITLE
fix(crypto): remove bound of Hash::try_from

### PR DESCRIPTION
**Pull request by wusyong**
# Description of change

Current `TryFrom` trait implementation of `Hash` has a trait bound that make it unable to convert any `Trits`/`TritBuf` to `Hash`. 
Remove it could make the conversion work again.

## Type of change

- Bug fix (a non-breaking change which fixes an issue)

## How the change has been tested

Tests under `bee-crypto`

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes
- [x] I have updated the CHANGELOG.md, if my changes are significant enough

### CODE REVIEW

File: bee-crypto/CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Deprecated
 
+- Fix `try_from` method of `Hash`
```
Comment (thibault-martinez) [inline bee-crypto/CHANGELOG.md]: Don't put it in the comment, create a new section

Comment (thibault-martinez) [reply bee-crypto/CHANGELOG.md]: I mean, if you want it released ASAP, otherwise yes, keep it there

File: bee-crypto/CHANGELOG.md [hunk 2]
```diff
@@ -19,6 +19,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Security -->
 
+## 0.1.2-alpha - 2020-08-19
+
+### Fixed
+
+- Fix `try_from` method of `Hash`
```
Comment (thibault-martinez) [inline bee-crypto/CHANGELOG.md]:
```suggestion
- `TryFrom<&'a Trits> for Hash`;
```

**Review by thibault-martinez — APPROVED** [review]

**Review by pvdrz — APPROVED** [review]

**Comment by thibault-martinez** [comment]
Obsolete due to cpt2.

**Pull request closed without merging**